OMCompiler/SimulationRuntime/c/simulation/solver/kinsolSolver.c

```diff
@@ -1203,14 +1203,16 @@ NLS_SOLVER_STATUS nlsKinsolSolve(DATA* data, threadData_t* threadData, NONLINEAR
         kinsolData->xScale,         /* scaling vector, for the variable cc */
         kinsolData->fScale);        /* scaling vector for function values fval */
 
-    if (flag < 0) {
+    if (flag < 0 && kinsolData->attemptRetry) {
       warningStreamPrint(LOG_NLS, 0, "KINSol finished with errorCode %d.", flag);
     } else {
       infoStreamPrint(LOG_NLS_V, 0, "KINSol finished with errorCode %d.", flag);
     }
     /* Try to handle recoverable errors */
     if (flag < 0 && kinsolData->attemptRetry) {
       retry = nlsKinsolErrorHandler(flag, data, nlsData, kinsolData);
+    } else {
+      retry = FALSE;
     }
```

Comment:
Less code = better code. What do you think?
```suggestion
    retry = flag < 0 && kinsolData->attemptRetry && nlsKinsolErrorHandler(flag, data, nlsData, kinsolData);
```

Reply:
Sounds like a good idea. I assume that nlsKinsolErrorHandler(flag, data, nlsData, kinsolData) is only called, if the first two conditions are true.

You can commit the changes, if you wish or do I have to do this?

Reply:
> Sounds like a good idea. I assume that nlsKinsolErrorHandler(flag, data, nlsData, kinsolData) is only called, if the first two conditions are true.

Yes, logical short circuiting should happen here.

> You can commit the changes, if you wish or do I have to do this?

I guess it's easiest if you do it. There should be a button somewhere.